app/src/main/java/co/touchlab/droidconandroid/shared/presenter/EventDetailViewModel.java

```diff
@@ -84,40 +88,34 @@ private void recordAnalytics(String analyticsKey, Event event) {
         );
     }
 
-    public void onEventMainThread(RemoveRsvpTask task) {
-        getDetails();
-    }
-
-    public void onEventMainThread(AddRsvpTask task) {
-        getDetails();
-    }
-
     public void unregister() {
         disposables.clear();
         host = null;
     }
 
-    // TODO: Replace with RSVPTasks that work
-//    public void toggleRsvp()
-//    {
-//        if(! ready())
-//        {
-//            return;
-//        }
-//
-//        long eventId = eventDetailInteractor.event.id;
-//        if(eventDetailInteractor.event.isRsvped())
-//        {
-//            TaskQueue.loadQueueDefault(getContext())
-//                    .execute(new RemoveRsvpTask(eventId));
-//            recordAnalytics(AnalyticsEvents.UNRSVP_EVENT);
-//        }
-//        else
-//        {
-//            TaskQueue.loadQueueDefault(getContext()).execute(new AddRsvpTask(eventId));
-//            recordAnalytics(AnalyticsEvents.RSVP_EVENT);
-//        }
-//    }
+    public void toggleRsvp(boolean isAlreadyRsvped) {
```

Comment:
OPTIONAL: isAlreadyRsvped is a slightly confusing way to control this. when I saw "toggleRsvp(true)" I thought it meant that you will be setting rsvp to true. I think its probably better to rename and reorganize this function so that its "setRsvp(boolean rsvp)" and calling it with true will add the rsvp